src/test/java/org/junit/tests/assertion/AssertionTest.java

```diff
@@ -674,4 +677,80 @@ public void assertNotEqualsIgnoresDeltaOnNaN() {
     public void assertNotEqualsIgnoresFloatDeltaOnNaN() {
         assertNotEquals(Float.NaN, Float.NaN, 1f);
     }
+
+    @Test(expected = AssertionError.class)
+    public void expectThrowsRequiresAnExceptionToBeThrown() {
+        expectThrows(nonThrowingRunnable());
+    }
+
+    @Test
+    public void expectThrowsIncludesNoMessageByDefault() {
+        try {
+            expectThrows(nonThrowingRunnable());
+            fail();
+        } catch (AssertionError ex) {
+            assertNull(ex.getMessage());
+        }
+    }
+
+    @Test
+    public void expectThrowsIncludesTheSuppliedMessage() {
+        try {
+            expectThrows("message", nonThrowingRunnable());
+            fail();
+        } catch (AssertionError ex) {
+            assertEquals("message", ex.getMessage());
+        }
+    }
+
+    @Test
+    public void expectThrowsReturnsTheSameObjectThrown() {
+        NullPointerException npe = new NullPointerException();
+
+        Throwable throwable = expectThrows(throwingRunnable(npe));
+
+        assertSame(npe, throwable);
+    }
+
+    @Test(expected = ClassCastException.class)
```

Comment:
I personally think throwing `ClassCastException` isn't very useful. You have to dig through the code (and possibly the implementation of `expectThrows()`) to understand why this was thrown. If the caller instead had to pass the exception type, the caller would always get a clear exception message.